File: services/pronounce.py
"""Rewrite brand names for TTS (avoid letter-by-letter NULLXES)."""

from __future__ import annotations

import re

_BRAND_RU = "Нуллксес"
_BRAND_EN = "Nullexes"
_NULLXES_RE = re.compile(r"NULLXES", re.IGNORECASE)


def for_speech(text: str) -> str:
    if not text:
        return text
    spoken = _BRAND_RU if re.search(r"[а-яё]", text, re.IGNORECASE) else _BRAND_EN
    return _NULLXES_RE.sub(spoken, text)

File: miniapp/src/lib/pronounce.ts
/** Rewrite brand names for TTS so engines don't spell them letter-by-letter. */

const BRAND_SPEECH_RU = "Нуллксес";
const BRAND_SPEECH_EN = "Nullexes";

function looksRussian(text: string): boolean {
  return /[а-яё]/i.test(text);
}

/** Text for Anam / TTS. Caption UI should keep the original spelling. */
export function forSpeech(text: string): string {
  if (!text) return text;
  const spoken = looksRussian(text) ? BRAND_SPEECH_RU : BRAND_SPEECH_EN;
  return text
    .replace(/NULLXES/gi, spoken)
    .replace(/Nullxes/g, spoken);
}
